multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ApplicationWaitAfterStopHandler.java

```diff
@@ -0,0 +1,52 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.time.Duration;
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.steps.ProcessContext;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+
+@Named
+public class ApplicationWaitAfterStopHandler {
+
+    protected ApplicationWaitAfterStopVariableGetter delayVariableGetter;
+
+    @Inject
+    public ApplicationWaitAfterStopHandler(ApplicationWaitAfterStopVariableGetter delayVariableGetter) {
+        this.delayVariableGetter = delayVariableGetter;
+    }
+
+    public boolean shouldWaitAfterAppStop(ProcessContext context) {
+        if (!delayVariableGetter.isAppStopDelayEnvVariableSet(context)) {
+            return false;
+        }
+        Duration durationToWait = delayVariableGetter.getDelayDurationFromAppEnv(context);
+        return !durationToWait.isZero();
+    }
+
+    public void setDelayAfterAppStop(ProcessContext context, String appName) {
+        boolean shouldWaitAfterAppStop = shouldWaitAfterAppStop(context);
+        context.setVariable(Variables.SHOULD_WAIT_AFTER_APP_STOP, shouldWaitAfterAppStop);
+        if (shouldWaitAfterAppStop) {
+            setWaitAfterAppStopVariable(context, appName);
+            context.getStepLogger()
+                   .info(Messages.DELAYING_APP_0_FOR_1_SECONDS, appName, context.getVariable(Variables.DELAY_AFTER_APP_STOP)
+                                                                                           .toSeconds());
+        }
+    }
+
+    private void setWaitAfterAppStopVariable(ProcessContext context, String appName) {
+        Duration durationToWait = delayVariableGetter.getDelayDurationFromAppEnv(context);
+        Duration currentDurationToWait = context.getVariable(Variables.DELAY_AFTER_APP_STOP);
```

Comment:
No, unless this method has been invoked from a previous process or operation, since this method is the only place that the variable is set currently.